tests/integration/support/client_wrapper.go

```diff
@@ -110,15 +111,19 @@ func GetInstance(apiOptions models.APIOptions) *ClientWrapper {
 	if err != nil {
 		log.Fatal(err)
 	}
-
+	trackListenerManager := optlyplugins.TrackListenerManager{}
+	for _, callback := range trackListenerManager.GetListenerCallbacks(apiOptions.Listeners) {
+		client.OnTrack(callback)
+	}
 	clientInstance = &ClientWrapper{
-		Client:             client,
-		DecisionService:    decisionService,
-		EventDispatcher:    eventProcessor.EventDispatcher,
-		UserProfileService: userProfileService,
-		OverrideStore:      overrideStore,
+		client:               client,
```

Comment:
Why this change is required in this PR. Lets have all these things in separate PR. This should focus only onTrack.